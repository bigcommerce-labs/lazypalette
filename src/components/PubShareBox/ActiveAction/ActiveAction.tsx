import React, {SFC} from 'react';

import ButtonInput from '../../ButtonInput/ButtonInput';
import { ButtonWrapper } from '../styles';

interface ActiveActionProps {
    isPrelaunchStore: boolean;
    loading: boolean;
    handlePublish(): void;
    handleSave(): void;
}

const ActiveAction: SFC<ActiveActionProps> = props => {
    return (
        <>
            {props.isPrelaunchStore ?
                <>
                    <ButtonWrapper>
                        <ButtonInput
                            onClick={props.handlePublish}
                            classType="primary"
                            type="button"
                            testId="save"
                            disabled={props.loading}
                        >
                            Save
                        </ButtonInput>
                    </ButtonWrapper>
                </> :
                <>
                    <ButtonWrapper>
                        <ButtonInput
                            onClick={props.handleSave}
                            type="button"
                            testId="save"
                            disabled={props.loading}
                        >
                            Save
                        </ButtonInput>
                    </ButtonWrapper>
                    <ButtonWrapper>
                        <ButtonInput
                            onClick={props.handlePublish}
                            classType="primary"
                            disabled={props.loading}
                            testId="publish"
                        >
                            Publish
                        </ButtonInput>
                    </ButtonWrapper>
                </>
            }
        </>
    );
};

export default ActiveAction;
