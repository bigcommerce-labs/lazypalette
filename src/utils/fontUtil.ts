export function parseFont(font: string) {
    const provider = font.split('_')[0];
    switch (provider) {
        case 'Google':
            return parseGoogleFont(font);
        default:
            return null;
    }
}

function parseGoogleFont(font: string) {
    const split = font.split('_');
    const family = split[1];
    let formattedFont = '';
    let weight = split[2];

    if (split.length === 2) {
        formattedFont += `${family}|`;
    } else if (split.length > 2) {
        weight = weight.split(',')[0];
        formattedFont += `${family}:${weight}|`;
    }

    return `//fonts.googleapis.com/css?family=${formattedFont}`;
}
