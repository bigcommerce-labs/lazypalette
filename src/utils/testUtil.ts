export const sel = (id: string) => `[data-test-id="${id}"]`;
