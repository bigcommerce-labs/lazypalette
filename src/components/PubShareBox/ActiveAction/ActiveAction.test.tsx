import { mount, shallow } from 'enzyme';
import React from 'react';

import { sel } from '../../../utils/testUtil';

import ActiveAction from './ActiveAction';

describe('ActiveAction', () => {

    const mockHandleSave = jest.fn(() => {
        activeActionMount.setProps({canSave: false});
    });
    const mockHandlePublish = jest.fn(() => {
        activeActionMount.setProps({canPublish: false});
    });

    const saveButton = sel('save');
    const publishButton = sel('publish');

    const activeActionShallow = shallow(<ActiveAction
        isPrelaunchStore={false}
        canPublish={false}
        canSave={false}
        handleSave={mockHandleSave}
        handlePublish={mockHandlePublish}
    />);

    const activeActionMount = mount(<ActiveAction
        isPrelaunchStore={false}
        canPublish={false}
        canSave={false}
        handleSave={mockHandleSave}
        handlePublish={mockHandlePublish}
    />);

    it('should render', () => {
        expect(activeActionShallow).toMatchSnapshot();
    });

    describe('when isPrelaunchStore is true', () => {
        it('should have only save button', () => {
            activeActionMount.setProps({isPrelaunchStore: true});
            expect(activeActionMount.find(saveButton).hostNodes()).toHaveLength(1);
        });
    });

    describe('when isPrelaunchStore is false', () => {
        it('should have both save and publish button', () => {
            activeActionMount.setProps({isPrelaunchStore: false});
            expect(activeActionMount.find(saveButton).hostNodes()).toHaveLength(1);
            expect(activeActionMount.find(publishButton).hostNodes()).toHaveLength(1);
        });
    });

    describe('save action', () => {
        it('should be enabled if canSave is true', () => {
            activeActionMount.setProps({canSave: true});
            expect(activeActionMount.find(saveButton).hostNodes().props().disabled).toBe(false);
        });

        it('should be disabled if canSave is false', () => {
            activeActionMount.setProps({canSave: false});
            expect(activeActionMount.find(saveButton).hostNodes().props().disabled).toBe(true);
        });

        describe('when canSave is true and save is clicked', () => {
            it('should call handleSave', () => {
                activeActionMount.setProps({canSave: true});
                activeActionMount.find(saveButton).hostNodes().simulate('click');
                expect(mockHandleSave).toHaveBeenCalledTimes(1);
            });
            it('should disable save button after', () => {
                expect(activeActionMount.find(saveButton).hostNodes().props().disabled).toBe(true);
            });
        });
    });

    describe('publish action', () => {
        it('should be enabled if canPublish is true', () => {
            activeActionMount.setProps({canPublish: true});
            expect(activeActionMount.find(publishButton).hostNodes().props().disabled).toBe(false);
        });

        it('should be disabled if canPublish is false', () => {
            activeActionMount.setProps({canPublish: false});
            expect(activeActionMount.find(publishButton).hostNodes().props().disabled).toBe(true);
        });

        describe('when canPublish is true and publish is clicked', () => {
            it('should call handlePublish', () => {
                activeActionMount.setProps({canPublish: true});
                activeActionMount.find(publishButton).hostNodes().simulate('click');
                expect(mockHandlePublish).toHaveBeenCalledTimes(1);
            });
            it('should disable publish button after', () => {
                expect(activeActionMount.find(publishButton).hostNodes().props().disabled).toBe(true);
            });
        });
    });
});
