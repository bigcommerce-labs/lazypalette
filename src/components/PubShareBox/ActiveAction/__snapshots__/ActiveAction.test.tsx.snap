// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveAction should render 1`] = `
<Fragment>
  <ButtonInput
    disabled={true}
    onClick={[MockFunction]}
    testId="save"
    type="button"
  >
    Save
  </ButtonInput>
  <ButtonInput
    classType="primary"
    disabled={true}
    onClick={[MockFunction]}
    testId="publish"
  >
    Publish
  </ButtonInput>
</Fragment>
`;
